fh-android-sdk/src/main/java/com/feedhenry/sdk/sync/FHSyncDataset.java

```diff
@@ -229,32 +249,30 @@ public void fail(FHResponse pResponse) {
     }
 
     private void syncRequestSuccess(JSONObject pData) {
-        // Check to see if any new pending records need to be updated to reflect the current state of play.
-        updatePendingFromNewData(pData);
-
         // Check to see if any previously crashed inflight records can now be resolved
         updateCrashedInFlightFromNewData(pData);
+        updateDelayedFromNewData(pData);
+        updateMetaFromNewData(pData);
 
-        // Update the new dataset with details of any inflight updates which we have not received a response on
-        updateNewDataFromInFlight(pData);
-
-        // Update the new dataset with details of any pending updates
-        updateNewDataFromPending(pData);
-
+        boolean hasRecords = false;
+        
         if (pData.has("records")) {
```

Comment:
this can be removed. As well as the resetDataset. It should be removed from the ios and wp sdk too.